### DISCUSSION TITLE
Where are the architectures implemented?

### DISCUSSION THREAD

**Discussion by noelfranthomas** [ggml-org/llama.cpp#4328] [Q&A] [answered]
Hello, I am very new to LLMs so please forgive my ignorance. I want to understand how these models are actually being implemented. So far, I understand that ggml helps with tensor ops, and the llama.cpp file has a lot of the workings of llama itself. Where are mistral and other llms implemented? Can we generate it from just the config file?

Any help would be greatly appreciated! My goal is to understand how llama.cpp works! Cheers!

**Comment by cmp-nct** [comment]
it's all in llama.cpp, just a historic name

**Comment by ggerganov** [comment, marked as answer]
See https://github.com/ggerganov/llama.cpp/blob/e4b76bbe316ee50fb17d9ac29e654c0edf830eba/llama.cpp#L5584-L5627

**Reply by noelfranthomas** [reply]
Thanks!
